pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantfiles/DownloadParticipantFileRoute.java

```diff
@@ -43,27 +49,28 @@ protected Object processRequest(Request request, spark.Response response, String
         String userIdR = UserUtil.getUserId(request);
         if (UserUtil.checkUserAccess(realm, userId, FILE_DOWNLOAD_ROLE, userIdR)) {
             String bucketName = queryParams.value(BUCKET);
-            String objectName = queryParams.value(FILE_NAME);
-            String blobName = queryParams.value(BLOB);
-            Optional<byte[]> downloadObject =
-                    downloadObject(googleProjectName, bucketName, objectName, blobName);
-            byte[] downloadObjectArray = downloadObject.orElseThrow();
-            response.status(200);
-            return downloadObjectArray;
+            String blobName = queryParams.value(BLOB_NAME);
+            String fileName = queryParams.value(FILE_NAME);
+            String fileGuid = queryParams.value(FILE_GUID);
+            String ddpParticipantId = queryParams.value(DDP_PARTICIPANT_ID);
+            Optional<DDPInstanceDto> ddpInstanceDto = new DDPInstanceDao().getDDPInstanceByInstanceName(realm);
+            ParticipantFilesUseCase participantFilesUseCase =
+                    new ParticipantFilesUseCase(bucketName, blobName, ddpParticipantId, fileGuid, fileDownloadService, ddpInstanceDto);
+
+            try {
+                SignedUrlResponse url = participantFilesUseCase.createSignedURLForDownload();
+                log.info("Signed URL generated for file download for participant " + ddpParticipantId + " file");
+                response.status(200);
+                return new Gson().toJson(url);
```

Comment:
once you add `new JsonTransformer()` while creating route in `DSMServer` you won't have to add `new Gson().toJson`